airbyte-integrations/connectors/destination-azure-blob-storage/src/main/java/io/airbyte/integrations/destination/azure_blob_storage/AzureBlobStorageConsumer.java

```diff
@@ -106,7 +106,14 @@ private void createContainers(final SpecializedBlobClientBuilder specializedBlob
       var blobItemList = StreamSupport.stream(containerClient.listBlobs().spliterator(), false)
           .collect(Collectors.toList());
       blobItemList.forEach(blob -> {
-        if (!blob.isDeleted() && blob.getName().contains(configuredStream.getStream().getName() + "/")) {
+        // Two important notes:
+        // 1. There's no option to write to a specific path, so we _always_ write to the root of the
+        // container.
+        // So the blob name always starts with `<stream_name>/`.
+        // 2. We really should include the namespace in the blob name, but we currently don't...
+        // So current behavior is that if you have `public1.users` and `public2.users`,
+        // those files will probably conflict in some way which is undesired.
+        if (!blob.isDeleted() && blob.getName().startsWith(configuredStream.getStream().getName() + "/")) {
```

Comment:
Is there any way to unit test this behavior?

Reply:
trying to figure that out :/ might need to write something into legacy DATs, which will be not very fun

Reply:
have a look at https://github.com/airbytehq/airbyte/pull/52610/commits/0886140ccfec26ae49a7f584cab031dd3c7713b8 ? It's modeled after the existing test stuff, which is doing something funky with an azure blob emulation container?

but it fails without the contains -> startsWith change, and passes afterward